### PR TITLE
dynamically set auto_tuned mode of WriteAmpBasedRateLimiter

### PR DESCRIPTION
**Pull request by tabokie**
Signed-off-by: tabokie <xy.tao@outlook.com>

Add new setter API to `RateLimiter`, only implement it for `WriteAmpBasedRateLimiter`.

### CODE REVIEW

File: utilities/rate_limiters/write_amp_based_rate_limiter.cc [hunk 1]
```diff
@@ -77,13 +77,33 @@ WriteAmpBasedRateLimiter::~WriteAmpBasedRateLimiter() {
...
+  if (auto_tuned_.load(std::memory_order_relaxed)) {
     max_bytes_per_sec_.store(bytes_per_second, std::memory_order_relaxed);
   } else {
     SetActualBytesPerSecond(bytes_per_second);
   }
 }
 
+void WriteAmpBasedRateLimiter::SetAutoTuned(bool auto_tuned) {
+  MutexLock g(&auto_tuned_mutex_);
+  if (auto_tuned_.load(std::memory_order_relaxed) != auto_tuned) {
```
Comment (yiwu-arbug) [inline utilities/rate_limiters/write_amp_based_rate_limiter.cc:89]: Let's use seq_cst order for all access to `auto_tuned_`?

**Review by yiwu-arbug — COMMENTED** [review]
rest LGTM

File: utilities/rate_limiters/write_amp_based_rate_limiter.cc (hunk 1 above)
Comment (tabokie) [reply utilities/rate_limiters/write_amp_based_rate_limiter.cc:89]: I think seq_cst is equivalent to acq_rel if only one variable is tagged. I went through the logic again and added some tags.

Comment (yiwu-arbug) [reply utilities/rate_limiters/write_amp_based_rate_limiter.cc]: yeah, using acquire/release ordering is even better, but sometimes they are hard to reason about. But as we are considering use arm, it make more sense for us to take some effort and use the acquire/release semantics if suitable.

**Review by yiwu-arbug — APPROVED** [review]
LGTM

**Comment by tabokie** [comment]
pin @Connor1996

**Review by Connor1996 — APPROVED** [review]

**Pull request merged**